cc/BUILD.bazel

```diff
@@ -29,6 +29,19 @@ config_setting(
     visibility = ["//visibility:public"],
 )
 
+# Enable warnings for deprecated declarations with --@rules_swiftnav//cc:warn_deprecated_declarations=true
+bool_flag(
```

Comment:
I like the PR overall, but this is making me nervous. This is exactly the thing that needs to be really annoying so it doesn't get forgotten and we end up in a situation where an API gets removed but we haven't updated code.

How about altering the `swift_cc_test` target so that it doesn't make use of this option, but all other target types do? It seems reasonable for unit tests to be interacting with deprecated and soon-to-be removed APIs, but we should remain annoyingly noisy on production code

Reply:
Completely agree.

I've only actually wired this config setting up in 'swift_cc_test' currently.

All the rest of the targets are getting compiled with -Wno-error=deprecated-declarations so the warnings are still there.

Maybe I should use a more descriptive name for this flag to make it more obvious it's only for tests?